### PR TITLE
Version 0.3.1

### PR DESCRIPTION
**Pull request by uphoffc**
* Bugfix: Add alias analysis for stack; needed to correctly insert barriers
* Bugfix: Disable block writes as alignment analysis is missing

### CODE REVIEW

File: src/visitor/aa_results.cpp [hunk 1]
```diff
@@ -8,15 +8,29 @@
...
+    : alias_(std::move(alias)), allocs_(std::move(allocs)) {}
 value_node *aa_results::root(value_node &a) {
     auto root = &a;
     if (alias_.find(root) != alias_.end()) {
         root = alias_[root];
     }
     return root;
 }
-bool aa_results::alias(value_node &a, value_node &b) { return root(a) == root(b); }
+bool aa_results::alias(value_node &a, value_node &b) {
```
Comment (hfp) [inline src/visitor/aa_results.cpp]: I guess expected signature is non-const for this binary function? Or perhaps `value_node` is const-qualified already.

File: src/gemm_generator.cpp [hunk 2]
```diff
@@ -348,9 +348,13 @@ void generator::add_microkernel(block_builder &bb, bool is_remainder, expr M, ex
         // 3. Data is adjacent in memory
         // 4. The address is 16 byte aligned
         // 5. We are not writing atomically
-        bool use_block_write =
+        /*bool use_block_write =
```
Comment (hfp) [inline src/gemm_generator.cpp]: More stylistic, but I would use `#if 0` ;-)

File: src/visitor/alias_analysis.cpp [hunk 3]
```diff
@@ -14,6 +16,14 @@ namespace tinytc {
 
 /* Stmt nodes */
 void alias_analyser::operator()(inst_node &) {}
+void alias_analyser::operator()(alloca_inst &a) {
```
Comment (hfp) [inline src/visitor/alias_analysis.cpp]: Perhaps const.

**Review by hfp — APPROVED** [review]
LGTM

File: src/visitor/aa_results.cpp (hunk 1 above)
Comment (uphoffc) [reply src/visitor/aa_results.cpp]: Changed to const

File: src/gemm_generator.cpp (hunk 2 above)
Comment (uphoffc) [reply src/gemm_generator.cpp]: ok

File: src/visitor/alias_analysis.cpp (hunk 3 above)
Comment (uphoffc) [reply src/visitor/alias_analysis.cpp]: Changed to const

**Pull request merged**